dns/dns_updater.go

```diff
@@ -0,0 +1,97 @@
+package dns
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/sky-uk/feed/controller"
+	"github.com/sky-uk/feed/dns/r53"
+	"github.com/sky-uk/feed/elb"
+	"k8s.io/heapster/Godeps/_workspace/src/github.com/coreos/fleet/log"
+)
+
+type updater struct {
+	dns          r53.Route53Client
+	elb          elb.ELB
+	elbLabelName string
+}
+
+// New creates an updater for dns
+func New(dnsClient r53.Route53Client, awsElb elb.ELB, elbLabelName string) controller.Updater {
+	return &updater{
+		dns:          dnsClient,
+		elb:          awsElb,
+		elbLabelName: elbLabelName,
+	}
+}
+
+func (u *updater) Start() error {
+
+	return nil
+}
+
+func (u *updater) Stop() error {
+
+	return nil
+}
+
+func (u *updater) Health() error {
+
+	return nil
+}
+
+func (u *updater) Update(update controller.IngressUpdate) error {
+	frontEnds, err := elb.FindFrontEndElbs(u.elb, u.elbLabelName)
```

Comment:
This needs to be cached or updated asynchronously. Like `feed-ingress` we should probably just query it at startup, with an `expected-frontends` parameter to ensure we find everything we expect. Otherwise this may hammer aws.